main.go

```diff
@@ -160,15 +164,16 @@ func runFetch(c *cli.Context) error {
 		return err
 	}
 
-	// Download the requested release asset
-	assetPath, err := downloadReleaseAsset(options.ReleaseAsset, options.LocalDownloadPath, repo, desiredTag)
+	// Download the requested release assets
+	assetPaths, err := downloadReleaseAssets(options.ReleaseAsset, options.LocalDownloadPath, repo, desiredTag)
 	if err != nil {
 		return err
 	}
 
 	// If applicable, verify the release asset
-	if options.ReleaseAssetChecksum != "" {
-		fetchErr = verifyChecksumOfReleaseAsset(assetPath, options.ReleaseAssetChecksum, options.ReleaseAssetChecksumAlgo)
+	if options.ReleaseAssetChecksum != "" && len(assetPaths) > 0 {
+		// TODO: Check more than just the first checksum if multiple assets were downloaded
+		fetchErr = verifyChecksumOfReleaseAsset(assetPaths[0], options.ReleaseAssetChecksum, options.ReleaseAssetChecksumAlgo)
```

Comment:
This behavior is a bit unintuitive. I would feel more comfortable if we error out the checksum check saying we don't support checksum checks if there are more than 1 assets matching the regex. However, I could use a second opinion on this. What do you think @josh-padnick ?

Reply:
I had the same thought. If the RegEx matches more than one asset, it seems like matching arbitrarily on the first of those matches is likely to lead to confusing behavior. If we really do have more than one asset path match the RegEx, I'd rather we return an error message with exactly what @yorinasub17 suggested: "Error: The regular expression <x> returned more than one match, and fetch only supports Regular Expressions that have exactly 1 one match."

I forget if fetch has any concept of debug logging, but as an added bonus, it'd be nice to show the user all the matches when in debug mode.

Reply:
@coryvirok actually, @josh-padnick and I talked again and we realized that maybe we can support checksum checks by supporting passing in multiple checksums via `--release-asset-checksum` and if the asset matches any of them, we pass the check.

If you can make the change to support this, that would be great! This should be possible (and backwards compatible) by using `StringSliceFlag` instead of `StringFlag` for `--release-asset-checksum`, which then allows passing in multiple entries. Then, you can use our `ListContainsElement` https://github.com/gruntwork-io/gruntwork-cli/blob/master/collections/lists.go#L4 function to check if the checksum list contains the asset checksum.